nntrainer/optimizers/adamw.cpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Daniel Jang <minhyukjang@snu.ac.kr>
+ *
+ * @file   adamw.cpp
+ * @date   3 November 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @author Daniel Jang <minhyukjang@snu.ac.kr>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is the AdamW Optimizer.
+ */
+
+#include <cmath>
+#include <fstream>
+
+#include <adamw.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+AdamW::AdamW() : adam_props(PropsB1(), PropsB2(), PropsEpsilon(), TorchRef()) {
+  /** default properties */
+  auto &[b1, b2, eps, torch_ref] = adam_props;
+  b1.set(0.9f);
+  b2.set(0.999f);
+  eps.set(1.0e-7f);
+  torch_ref.set(false);
+}
+
+AdamW::~AdamW() {}
+
+enum AdamParams { wm, wv };
+
+std::vector<TensorDim> AdamW::getOptimizerVariableDim(const TensorDim &dim) {
+  return {dim, dim};
+}
+
+void AdamW::exportTo(Exporter &exporter,
+                     const ml::train::ExportMethods &method) const {
+  exporter.saveResult(adam_props, method, this);
+  Optimizer::exportTo(exporter, method);
+}
+
+void AdamW::setProperty(const std::vector<std::string> &values) {
+  auto left = loadProperties(values, adam_props);
+  Optimizer::setProperty(left);
+}
+
+void AdamW::applyGradient(RunOptimizerContext &context) {
+  Tensor &x_grad = context.getGradient();
```

Comment:
Code to process mixed precision is missing. Please refer to 
https://github.com/nnstreamer/nntrainer/blob/0d74457e18b97fa8d26702ec7663d6f6e9510ac2/nntrainer/optimizers/adam.cpp#L75-L84

Reply:
yes. this needs to be considered. Please check it.

Reply:
This, too, has been addressed.